plugins/atomicfu/atomicfu-compiler/src/org.jetbrains.kotlinx.atomicfu.compiler/extensions/AtomicfuTransformer.kt

```diff
@@ -101,6 +103,44 @@ class AtomicfuTransformer(private val context: IrPluginContext) {
 
     private inner class AtomicTransformer : IrElementTransformer<IrFunction?> {
 
+        override fun visitProperty(declaration: IrProperty, data: IrFunction?): IrStatement {
+            // Support transformation for delegated properties:
+            if (declaration.isDelegated && declaration.backingField?.type?.isAtomicValueType() == true) {
+                declaration.backingField?.let { delegateBackingField ->
+                    delegateBackingField.initializer?.let {
+                        val initializer = it.expression as IrCall
+                        when {
+                            initializer.isAtomicFieldGetter() -> {
+                                // val _a = atomic(0)
+                                // var a: Int by _a
+                                // Accessors of the delegated property `a` are implemented via the generated property `a$delegate`,
+                                // that is the copy of the original `_a`.
+                                // They should be delegated to the value of the original field `_a` instead of `a$delegate`.
+
+                                // fun <get-a>() = a$delegate.value -> _a.value
+                                // fun <set-a>(value: Int) = { a$delegate.value = value } -> { _a.value = value }
+                                val initializerName = initializer.symbol.owner.name.asString()
+                                val originalFieldName = initializerName.getFieldName()
+                                val originalProperty = (declaration.parent as IrDeclarationContainer).declarations.single {
```

Comment:
What if
```kotlin
val a = atomic(42)
class B {
  val b: Int by a
}
```

Reply:
Probably `call.symbol.owner.corresponidingProperty` would help

Reply:
Fixed this. `originalField` is just the backing field of the initializer and I can get it via `corresponidingProperty`.

`originalField.backingField()`